capture/d2l-capture-producer/d2l-capture-producer-editor.js

```
@@ -921,13 +927,13 @@ class CaptureProducerEditor extends RtlMixin(InternalLocalizeMixin(LitElement))
 			pixelsAlongTimelineToZoomAround: this._timeline?.pixelsAlongTimelineToZoomAround
 		});
 
-		for (const cut of this._timeline.getCutsOnTimeline()) this._addCutToStage(cut);
-
-		this._updateMouseEnabledForCuts();
-
 		for (const mark of this._timeline.getMarksOnTimeline()) this._addMarkToStage(mark);
 
 		this._updateMouseEnabledForMarks();
+
+		for (const cut of this._timeline.getCutsOnTimeline()) this._addCutToStage(cut);
+
+		this._updateMouseEnabledForCuts();
```

Timeline Cuts are [added by index](https://github.com/Brightspace/content-components/blob/7e1a4ed77d00bf6e900aa496a234335a640aa5e9/capture/d2l-capture-producer/d2l-capture-producer-editor.js#L262) so that their indices are lower than Timeline Marks. This is to ensure that Marks are drawn on top of Cuts on the timeline graphic.

Here is the line in `_addCutToStage()` that adds the Cut by index:

```
this._stage.addChildAt(displayObject, this._stage.numChildren - this._timeline.getMarksOnTimeline().length - 1);
```

Note that the index is calculated based on the number of Timeline Cuts. However [`this._stage.addChildAt()` silently fails if the index is larger than the number of graphics currently added to the timeline](https://www.createjs.com/docs/easeljs/classes/Stage.html). Since the old code added the Cuts to `this._stage` _before_ adding the Marks, this caused the timeline graphic to break.

The fix was to swap the ordering so that Marks are added to `this._stage` first, followed by the Cuts.